File: js/anagram2.js
anagram = function(string1, string2) {
    
    let arr1 = [], arr2 = []

    for(let i of string1.toLowerCase()){
        if( i !== ' '){
            arr1.push(i)
        }
    }

    for(let j of string2.toLowerCase()){
        if( j !== ' '){
            arr2.push(j)
        }
    }
    
    str1 = arr1.sort().join('')
    str2 = arr2.sort().join('')
    
    return str1 === str2
}

exports.anagramsFor = function(word, listOfWords) {
    
    let answer = []
 
    for(let i of listOfWords){
        if(anagram(word, i)){
            answer.push(i)
        }
    }

    return answer

}

File: python/anagram2.py
def anagram(string1, string2):
	list1 = []
	list2 = []

	for i in sorted(string1.lower()):
		if i != ' ':
			list1.append(i)

	for j in sorted(string2.lower()):
		if j != ' ':
			list2.append(j)
	
	str1 = ''.join(list1)
	str2 = ''.join(list2)

	return str1 == str2

def anagrams_for(word, list_of_words):
	answer = []

	for i in list_of_words:
		if anagram(word, i):
			answer.append(i)
	return answer
